### PR TITLE
NIFI-7436 Ability to walk Record FieldValue to root

### PR DESCRIPTION
**Pull request by ottobackwards**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

This pr adds utility classes to walk the FieldValue hierarchy from any given node to root, and build a logical path as a result.

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [-] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [-] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [-] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [-] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [-] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [-] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

---
This change is [Reviewable](https://reviewable.io/reviews/apache/nifi/4262)

### CODE REVIEW

**Comment by ottobackwards** [comment]
any chance to get a review on this?

**Comment by ottobackwards** [comment]
touch

**Comment by ottobackwards** [comment]
Happy New Year!

**Comment by markap14** [comment]
Happy New Year @ottobackwards ! Should be able to review this this week.

**Review by markap14 — COMMENTED** [review]
Thanks @ottobackwards, code looks good. Pretty thoroughly unit tested. The only feedback I would have was commented inline. The repetitive calls to `StringBuilder.insert(0, <string>);` are concerning. I think there are good alternatives to that. Otherwise, all good from my perspective.

File: nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/util/FieldValueLogicalPathBuilder.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+     * Builds a logical path String using the configured wrappers for array or map values for a given
+     * {@code FieldValue}
+     *
+     * @param fieldValue the Field Value
+     * @return a String with a path
+     */
+    public String buildLogicalPath(FieldValue fieldValue) {
+        Objects.requireNonNull(fieldValue, "fieldValue cannot be null");
+        final StringBuilder stringBuilder = new StringBuilder();
+        FieldValueWalker.walk(fieldValue, (thisFieldValue) -> {
```
[the remark this answers is not in the document]
Comment (ottobackwards) [reply nifi-commons/nifi-record-path/src/main/java/org/apache/nifi/record/path/util/FieldValueLogicalPathBuilder.java:103]: thanks for the review, let me look at it and try to remember what the hell i was doing.  It has been a bit lol.

**Comment by markap14** [comment]
Thanks @ottobackwards! Looks good to me. Merged to main!

**Pull request closed without merging**